great_tables/_spanners.py

```diff
@@ -565,26 +565,49 @@ def empty_spanner_matrix(
     return [{var: var for var in vars}], vars
 
 
-def seq_groups(seq: list[str]):
+def pairwise(iterable: Iterable[Any]) -> Generator[tuple[Any, Any], None, None]:
+    """
+    https://docs.python.org/3/library/itertools.html#itertools.pairwise
+    pairwise('ABCDEFG') → AB BC CD DE EF FG
+    """
+    # This function can be replaced by `itertools.pairwise` if we only plan to support
+    # Python 3.10+ in the future.
+    iterator = iter(iterable)
+    a = next(iterator, None)
+    for b in iterator:
+        yield a, b
+        a = b
```

Comment:
Is this `a = b` a typo?

Reply:
I believe it's not. Let's take "ABCDEFG" as an example. First, we get the first element, `a`, which is "A". At this point, "BCDEFG" remains in the iterator. 

Next, we iterate through the iterator and get the next element, `b`, which is "B", leaving "CDEFG" in the iterator. We then yield the tuple ("A", "B") and set `a` to `b`, so `a` now equals "B".

We continue iterating through the iterator. The next element is "C", with "DEFG" remaining in the iterator. We yield the tuple ("B", "C"), and the process repeats in the same manner.

Reply:
ah shoot--that makes sense! Sorry, I spaced very briefly on how iterators work 😓